libs/go/sia/host/utils/utils.go

```diff
@@ -44,3 +48,55 @@ func GetHostname(fqdn bool) string {
 		return hostname
 	}
 }
+
+// GetK8SHostnames Generate pod/svc hostnames based on k8s spec:
+// https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#pods
+func GetK8SHostnames() []string {
+	k8sDnsEntries := []string{}
+	// we're going to generate two additional sanDNS entries for our
+	// instances running with K8S - pod and service entries. it requires
+	// that the container was configured with the expected env values
+	// using the downward API.
+	// the pod name and namespace may already be available through other
+	// variables/settings, but we'll give preference to our env settings
+	// if they have been configured.
+	podNamespace := os.Getenv("ATHENZ_SIA_POD_NAMESPACE")
+	if podNamespace == "" {
+		// use namespace associated with the service account
+		if ns, err := os.ReadFile("/var/run/secrets/kubernetes.io/serviceaccount/namespace"); err == nil {
+			podNamespace = strings.TrimSpace(string(ns))
+		}
+	}
+	// in all of our components we need our namespace so if we don't
+	// have one configured there is no need to fetch other settings
+	if podNamespace == "" {
+		return k8sDnsEntries
+	}
+
+	podName := os.Getenv("ATHENZ_SIA_POD_NAME")
+	if podName == "" {
+		podName = os.Getenv("HOSTNAME")
+	}
+	podIP := os.Getenv("ATHENZ_SIA_POD_IP")
+	podService := os.Getenv("ATHENZ_SIA_POD_SERVICE")
+	podSubdomain := os.Getenv("ATHENZ_SIA_POD_SUBDOMAIN")
+
+	if podIP != "" {
+		podIPWithDashes := strings.ReplaceAll(podIP, ".", "-")
```

Comment:
do we have to worry about IPv6?

Reply:
I haven't seen any references about using IPv6 in the pod dns name. So the expectation is that you'll set the ATHENZ_SIA_POD_IP only if you have ipv4 assigned to your pod otherwise, you won't get the ip based pod dns name in your sandns list.